File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/utils/StringUtils.java
package com.raizlabs.android.dbflow.processor.utils;

/**
 * Description:
 */
public class StringUtils {

    public static boolean isNullOrEmpty(String str) {
        return str == null || str.trim().length() == 0 || str.equals("null");
    }

    public static String capitalize(String str) {
        if (str == null || str.trim().length() == 0) {
            return str;
        }

        return str.substring(0, 1).toUpperCase() + str.substring(1);
    }

    public static String lower(String str) {
        if (str == null || str.trim().length() == 0) {
            return str;
        }

        return str.substring(0, 1).toLowerCase() + str.substring(1);
    }
}

File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/utils/StringUtils.kt
package com.raizlabs.android.dbflow.processor.utils

/**
 * Description:
 */
fun String?.isNullOrEmpty(): Boolean {
    return this == null || this.trim { it <= ' ' }.isEmpty() || this == "null"
}

fun String?.capitalizeFirstLetter(): String {
    if (this == null || this.trim { it <= ' ' }.isEmpty()) {
        return this ?: ""
    }

    return this.capitalize()
}

fun String?.lower(): String {
    if (this == null || this.trim { it <= ' ' }.isEmpty()) {
        return this ?: ""
    }

    return this.substring(0, 1).toLowerCase() + this.substring(1)
}
